hdf5-types/src/array.rs

```diff
@@ -14,76 +14,22 @@ pub unsafe trait Array: 'static {
     fn capacity() -> usize;
 }
 
```

Comment:
I think this trait `Array` is now not needed at all. The idea was initially borrowed from `arrayvec` which now also switched to const generics - see https://github.com/bluss/arrayvec/pull/172 (and the removed `array.rs`)?